src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -460,26 +441,29 @@ internal static string DecodeStream(Stream stream, ref Encoding encoding)
             string content = StreamToString(stream, encoding);
             if (isDefaultEncoding)
             {
-                do
+                // Check for a charset attribute on the meta element to override the default
+                // we only look within the first 1k characters as the meta tag is in the head
+                // tag which is at the start of the document
+                Match match = s_metaRegex.Match(content.Substring(0, Math.Min(content.Length, 1024)));
+                
+                // Check for a encoding attribute on the xml declaration to override the default
+                // we only look within the first 256 characters as the declaration is in the first
+                // line of the document
+                Match match2 = s_xmlRegex.Match(content.Substring(0, Math.Min(content.Length, 256)));
```

Comment:
I think we can reuse `content.Substring` from previous statement.

Reply:
The XML declaration must be the first construct in the document, because of this I decided to limit the search to the first 256 characters. I will change it to 1024

Reply:
Done